### ISSUE TITLE
Not fixed: Failed to run serialization weaver with command "Temp\StagingArea\Data\Managed\Newtonsoft.Json.dll"

### ISSUE THREAD

**Issue by ryamldess** [JamesNK/Newtonsoft.Json#1031]
In an identical issue, I tried the suggestion of the person who opened, and subsequently closed it, of using the .dll in this fork: https://github.com/SaladLab/Json.Net.Unity3D/releases/download/8.0.3/JsonNet-UwpWorkaround.8.0.3.zip.

It did not work. Attempting to compile against the Universal Windows 10 target with D3D/Local Machine in the Unity HoloLens technical preview, I get the same error. As a substantial portion of my underlying library code relies on this .dll, this is currently blocking me from building for HoloLens.

Here is the error:

UnityException: Failed to run serialization weaver with command "Temp\StagingArea\Data\Managed\Newtonsoft.Json.dll" -pdb -verbose -unity-engine="Temp\StagingArea\Data\Managed\UnityEngine.dll" "Temp\StagingArea\TempSerializationWeaver" -lock=UWP\project.lock.json -additionalAssemblyPath="Temp\StagingArea\Data\Managed" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\UnionMetadata\Facade" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.Calls.CallsVoipContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.SocialInfo.SocialInfoContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Devices.DevicesLowLevelContract\2.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Devices.Printers.PrintersContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Foundation.FoundationContract\2.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Foundation.UniversalApiContract\3.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Graphics.Printing3D.Printing3DContract\3.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Networking.Connectivity.WwanContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Services.Store.StoreContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.System.Profile.ProfileHardwareTokenContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.System.Profile.ProfileSharedModeContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.UI.ViewManagement.ViewManagementViewScalingContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.Activation.ActivatedEventsContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.Activation.ActivationCameraSettingsContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.Activation.ContactActivatedEventsContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.Activation.WebUISearchActivatedEventsContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.Background.BackgroundAlarmApplicationContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.Calls.Background.CallsBackgroundContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.Calls.LockScreenCallContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.FullTrustAppContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.Preview.Notes.PreviewNotesContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.Resources.Management.ResourceIndexerContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.Search.Core.SearchCoreContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.Search.SearchContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.StartupTaskContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.Wallet.WalletContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Devices.Custom.CustomDeviceContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Devices.Portable.PortableDeviceContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Devices.Printers.Extensions.ExtensionsContract\2.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Devices.Scanners.ScannerDeviceContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Devices.Sms.LegacySmsApiContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Gaming.Preview.GamesEnumerationContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Globalization.GlobalizationJapanesePhoneticAnalyzerContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Management.Deployment.Preview.DeploymentPreviewContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Management.Orchestration.OrchestrationContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Management.Workplace.WorkplaceSettingsContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Media.Capture.AppCaptureContract\3.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Media.Capture.CameraCaptureUIContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Media.Devices.CallControlContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Media.MediaControlContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Media.Playlists.PlaylistsContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Media.Protection.ProtectionRenewalContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Networking.NetworkOperators.LegacyNetworkOperatorsContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Networking.Sockets.ControlChannelTriggerContract\2.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Perception.Automation.Core.PerceptionAutomationCoreContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Security.EnterpriseData.EnterpriseDataContract\3.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Security.ExchangeActiveSyncProvisioning.EasContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Services.Maps.GuidanceContract\2.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Services.Maps.LocalSearchContract\3.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.System.Profile.SystemManufacturers.SystemManufacturersContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.System.Profile.ProfileRetailInfoContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.System.UserProfile.UserProfileContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.System.UserProfile.UserProfileLockScreenContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.UI.ApplicationSettings.ApplicationsSettingsContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.UI.Core.AnimationMetrics.AnimationMetricsContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.UI.Core.CoreWindowDialogsContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.UI.Xaml.Hosting.HostingContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Web.Http.Diagnostics.HttpDiagnosticsContract\2.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.System.SystemManagementContract\2.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.Calls.CallsPhoneContract\3.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.ApplicationModel.CommunicationBlocking.CommunicationBlockingContract\2.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Devices.SmartCards.SmartCardBackgroundTriggerContract\2.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Devices.SmartCards.SmartCardEmulatorContract\3.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Embedded.DeviceLockdown.DeviceLockdownContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Networking.NetworkOperators.NetworkOperatorsFdnContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Phone.PhoneContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.Phone.StartScreen.DualSimTileContract\1.0.0.0" -additionalAssemblyPath="C:\Program Files (x86)\Windows Kits\10\References\Windows.UI.WebUI.Core.WebUICommandBarContract\1.0.0.0".
Symbols will be read from Temp\StagingArea\Data\Managed\UnityEngine.pdb
No symbols for Temp\StagingArea\Data\Managed\Newtonsoft.Json.dll
Weaving assembly C:\Users\stevesedlmayr\Desktop\Beelzebub's Bubble Toy Bonanza\source\hg\droidknot-bbtb-unity\code\BBTB-Unity\Temp\StagingArea\Data\Managed\Newtonsoft.Json.dll
 + Newtonsoft.Json.JsonException
 + Newtonsoft.Json.JsonReaderException
 + Newtonsoft.Json.JsonSerializationException
 + Newtonsoft.Json.JsonWriterException
Will export symbols of pdb format
Mono.Cecil.ResolutionException: Failed to resolve System.Runtime.Serialization.Formatters.FormatterAssemblyStyle
   at Mono.Cecil.Mixin.CheckedResolve(TypeReference self)
   at Mono.Cecil.MetadataBuilder.GetConstantType(TypeReference constant_type, Object constant)
   at Mono.Cecil.MetadataBuilder.AddConstant(IConstantProvider owner, TypeReference type)
   at Mono.Cecil.MetadataBuilder.AddField(FieldDefinition field)
   at Mono.Cecil.MetadataBuilder.AddFields(TypeDefinition type)
   at Mono.Cecil.MetadataBuilder.AddType(TypeDefinition type)
   at Mono.Cecil.MetadataBuilder.AddTypeDefs()
   at Mono.Cecil.MetadataBuilder.BuildTypes()
   at Mono.Cecil.MetadataBuilder.BuildModule()
   at Mono.Cecil.ModuleWriter.<BuildMetadata>b__0(MetadataBuilder builder, MetadataReader _)
   at Mono.Cecil.ModuleDefinition.Read[TItem,TRet](TItem item, Func`3 read)
   at Mono.Cecil.ModuleWriter.WriteModuleTo(ModuleDefinition module, Stream stream, WriterParameters parameters)
   at Mono.Cecil.ModuleDefinition.Write(String fileName, WriterParameters parameters)
   at usw.Weaver.WeaveAssembly(String assemblyPath, AssemblyDefinition unityEngineAssemblyDefinition, ReaderParameters readerParameters)
   at usw.Weaver.Weave()
   at usw.Program.RunProgram(ConversionOptions options)
   at usw.Program.Main(String[] args)

PostProcessWinRT.RunSerializationWeaver () (at C:/buildslave/unity/build/PlatformDependent/WinRT/SharedSources/CSharp/PostProcessWinRT.cs:465)
PostProcessWinRT.Process () (at C:/buildslave/unity/build/PlatformDependent/WinRT/SharedSources/CSharp/PostProcessWinRT.cs:123)
UnityEditor.Metro.BuildPostprocessor.PostProcess (BuildPostProcessArgs args) (at C:/buildslave/unity/build/PlatformDependent/MetroPlayer/Extensions/Managed/ExtensionModule.cs:130)
UnityEditor.PostprocessBuildPlayer.Postprocess (BuildTarget target, System.String installPath, System.String companyName, System.String productName, Int32 width, Int32 height, System.String downloadWebplayerUrl, System.String manualDownloadWebplayerUrl, BuildOptions options, UnityEditor.RuntimeClassRegistry usedClassRegistry, UnityEditor.BuildReporting.BuildReport report) (at C:/buildslave/unity/build/Editor/Mono/BuildPipeline/PostprocessBuildPlayer.cs:176)

**Comment by ryamldess** [comment]
I should mention that I also tried the portable40, net20 and net35 builds before trying the UWP fork and they all failed as well.

**Comment by JamesNK** [comment]
I don't know anything about Unity. Shouldn't you raise this bug with whoever made Json.Net.Unity3D?

**Comment by ryamldess** [comment]
James, I think you misunderstood. This error emanates from **your** code. The Json.Net.Unity3D version was a workaround that I tried at the suggestion of someone reporting the same bug with **your** code. The poster closed that issue as resolved due to this workaround existing. However, the workaround does not work; it throws the same error as your own original code. I included it as it is customary when reporting issues to include steps you have tried to rectify the issue.
